linode/manager
# feat: [LKEAPIFW-428] LKE clusters should have IP ACL integration on CM (part 1)

## Description 📝
Enables creation of LKE clusters with IP ACLS; enables updates of already migrated clusters

## Changes  🔄
- Adds new UI components
- adds respective library calls
- some style cleanup

*** tests will be in separate PRs: #11131 and #11132 

## Target release date 🗓️
Oct 28th release

## Preview 📷
Cluster creation:

<img src="https://github.com/user-attachments/assets/27b9aea7-cbc6-4e18-bb8a-b3382543e7b6" width="550" />

Cluster summary:
| IPACL enabled  | IPACL not enabled   |
| ------- | ------- |
| ![Screenshot 2024-09-04 at 14 47 05](https://github.com/user-attachments/assets/47defaf4-1fb3-4fe3-a906-3445a4a9ba87) | ![Screenshot 2024-10-08 at 18 28 48](https://github.com/user-attachments/assets/3331f7c0-4af6-4eed-95f1-68d7c9a7eeb2) |

New IPACL Drawer:
| Cluster migrated (IPACL enabled/not enabled) | Cluster not yet migrated   |
| ------- | ------- |
| ![Screenshot 2024-09-04 at 14 47 05](https://github.com/user-attachments/assets/ce48ffbb-f905-405b-86ab-c82753b75c39) | ![Screenshot 2024-10-08 at 18 28 48](https://github.com/user-attachments/assets/de483efa-5506-43fd-a983-84d70ecaa50e) |

## How to test 🧪

Spin up this branch, update to reach alpha's apinext backend. Once all set, attempt to create new clusters with ipacls and then update them.

### Prerequisites
See how to test

### Verification steps
- Create a Cluster with IPACL enabled (toggled on). Confirm button on summary page lists number of IPs added
- Create a Cluster with IPACL disabled (toggled off). Confirm button on summary page says 'Enable'. If you added any IPs, confirm they show up in the drawer
- Create a Cluster that does not have IPACL. Confirm button on summary page says 'Enable' (it will take several seconds to load) and that drawer matches the screenshot for 'Cluster not yet migrated'. Update/toggle IPACL, confirm flow now follows a cluster with IPACL enabled
  - See Talmai's comment for how to create these clusters. I'v also just been creating clusters on alpha (but not on this branch/using the develop branch), and then going back to this branch
  - general spotcheck of Kubernetes with both the APL flag on and off

## As an Author I have considered 🤔

*Check all that apply*

- ✅ 👀 Doing a self review
- ✅ ❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
- ✅ 🤏 Splitting feature into small PRs
- [NO see comment] ➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
- [NO ] 🧪 Providing/Improving test coverage
- ✅ 🔐 Removing all sensitive information from the code and PR description
- [NO] 🚩 Using a feature flag to protect the release
- ✅ 👣 Providing comprehensive reproduction steps
- [NO] 📑 Providing or updating our documentation
- [NO] 🕛 Scheduling a pair reviewing session
- [N/A] 📱 Providing mobile support
- [N/A] ♿  Providing accessibility support
